packages/block-editor/src/components/block-title/use-block-display-title.js

```diff
@@ -35,49 +34,52 @@ export default function useBlockDisplayTitle( {
 	maximumLength,
 	context,
 } ) {
-	const { attributes, name, reusableBlockTitle } = useSelect(
+	const blockTitle = useSelect(
 		( select ) => {
 			if ( ! clientId ) {
-				return {};
+				return null;
 			}
+
 			const {
 				getBlockName,
 				getBlockAttributes,
 				__experimentalGetReusableBlockTitle,
 			} = select( blockEditorStore );
+			const { getBlockType, getActiveBlockVariation } =
+				select( blocksStore );
+
 			const blockName = getBlockName( clientId );
-			if ( ! blockName ) {
-				return {};
+			const blockType = getBlockType( blockName );
+			if ( ! blockType ) {
+				return null;
 			}
-			const isReusable = isReusableBlock( getBlockType( blockName ) );
-			return {
-				attributes: getBlockAttributes( clientId ),
-				name: blockName,
-				reusableBlockTitle:
-					isReusable &&
-					__experimentalGetReusableBlockTitle(
-						getBlockAttributes( clientId ).ref
-					),
-			};
+
+			const attributes = getBlockAttributes( clientId );
+			const isReusable = isReusableBlock( blockType );
+			const reusableBlockTitle = isReusable
+				? __experimentalGetReusableBlockTitle( attributes.ref )
```

Comment:
We should probably deprecate this selector and handle the pattern block title via `__experimentalLabel`.

cc @talldan, @kevin940726

Reply:
Note that I'm adjusting this selector here [Avoid fetching ALL reusable blocks (user patterns) on post/site editor load](https://github.com/WordPress/gutenberg/pull/58250/files#top)

Reply:
Here's the correct PR link - https://github.com/WordPress/gutenberg/pull/58239.